pkg/report/handler.go

```diff
@@ -28,3 +35,45 @@ func ShowHandler() func(w http.ResponseWriter, r *http.Request) {
 		http.FileServer(fs).ServeHTTP(w, r)
 	}
 }
+
+//PersistHandler method streams request to storage presigned URL
+func PersistHandler(kubeClient *kk8s.Client) func(w http.ResponseWriter, r *http.Request) {
+	if minioClient == nil {
+		panic("client was not initialized, please initialize object storage client")
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		loadTestName := chi.URLParam(r, "id")
+
+		_, err := kubeClient.GetLoadTest(r.Context(), loadTestName)
+		if errors.IsNotFound(err) {
+			render.Render(w, r, khttp.ErrResponse(http.StatusNotFound, err.Error()))
+			return
+		}
+		if err != nil {
+			render.Render(w, r, khttp.ErrResponse(http.StatusInternalServerError, err.Error()))
+			return
+		}
+
+		url, err := newPreSignedPutURL(loadTestName)
+		if nil != err {
+			render.Render(w, r, khttp.ErrResponse(http.StatusInternalServerError, err.Error()))
+			return
+		}
+
+		proxyReq, _ := http.NewRequest(r.Method, url.String(), r.Body)
```

Comment:
we should use `NewRequestWithContext` and setup cancel timeout

Reply:
I change to use the parent request, as I literally streaming it